VpNet/ManagedApi/VirtualParadiseClient.NativeEvents.cs

```diff
@@ -0,0 +1,575 @@
+
+            GetVpObject(sender, out VpObject vpObject);
+
+            if (session == 0)
+                QueryCellResult?.Invoke(this, new QueryCellResultArgs(vpObject));
+            else
+                ObjectCreated?.Invoke(this, new ObjectCreateArgs(avatar, vpObject));
+        }
+
+        private void OnObjectChangeNative(IntPtr sender)
+        {
+            if (ObjectChanged == null) return;
+            VpObject vpObject;
+            int sessionId;
+            lock (this)
+            {
+                GetVpObject(sender, out vpObject);
+                sessionId = Functions.vp_int(sender, IntegerAttribute.AvatarSession);
+            }
+
+            ObjectChanged(this, new ObjectChangeArgs(GetAvatar(sessionId), vpObject));
+        }
+
+        private void OnQueryCellEndNative(IntPtr sender)
+        {
+            if (QueryCellEnd == null) return;
+            int x;
+            int z;
+            lock (this)
+            {
+                x = Functions.vp_int(sender, IntegerAttribute.CellX);
+                z = Functions.vp_int(sender, IntegerAttribute.CellZ);
+            }
+
+            QueryCellEnd(this, new QueryCellEndArgs(new Cell(x, z)));
+        }
+
+        private void OnWorldListNative(IntPtr sender)
+        {
+            if (WorldListEntryReceived == null)
+                return;
+
+            World data;
+            lock (this)
+            {
+                string worldName = Functions.vp_string(NativeInstanceHandle, StringAttribute.WorldName);
+                data = new World()
+                {
+                    Name = worldName,
+                    State = (WorldState) Functions.vp_int(NativeInstanceHandle, IntegerAttribute.WorldState),
+                    UserCount = Functions.vp_int(NativeInstanceHandle, IntegerAttribute.WorldUsers)
+                };
+            }
+
+            if (_worlds.ContainsKey(data.Name))
+                _worlds.Remove(data.Name);
+            _worlds.Add(data.Name, data);
+            WorldListEntryReceived(this, new WorldListEventArgs(data));
+        }
+
+        private void OnWorldSettingNative(IntPtr instance)
+        {
+            if (!_worlds.ContainsKey(Configuration.World.Name))
+            {
+                _worlds.Add(Configuration.World.Name, Configuration.World);
+            }
+
+            var world = _worlds[Configuration.World.Name];
+            var key = Functions.vp_string(instance, StringAttribute.WorldSettingKey);
+            var value = Functions.vp_string(instance, StringAttribute.WorldSettingValue);
+            world.RawAttributes[key] = value;
+        }
+
+        private void OnWorldSettingsChangedNative(IntPtr instance)
+        {
+            // Initialize World Object Cache if a local object path has been specified and a objectpath is speficied in the world attributes.
+            // TODO: some world, such as Test do not specify a objectpath, maybe there's a default search path we dont know of.
```

Comment:
I guess these comments are about something that no longer exists.

Reply:
I don't recall writing these comments, I think they were already there. However I'm thinking of adding a more intuitive way of handling world settings 🤔